django/contrib/sessions/backends/signed_cookies.py

```diff
@@ -79,3 +97,7 @@ def _get_session_key(self):
     @classmethod
     def clear_expired(cls):
         pass
+
+    @classmethod
+    async def aclear_expired(cls):
+        pass
```

Comment:
Don't you think that we should do the same for file-based backend to avoid context switching?

Reply:
@bigfootjon What do you think? :point_up:

Reply:
Is there an async API for files? My understanding is that Python does not expose an async file io model. IIRC somewhere in Django (ASGI support for static files?) that has a context switch due to this same limitation.

Reply:
There is `aiofiles` https://pypi.org/project/aiofiles/

Reply:
I figured this wasn’t the right vehicle to take on a new dependency, I think aiofiles would be a larger conversation

Reply:
I didn't want to suggest that :) I think it's worth adding for consistency and to avoid unnecessary `sync_to_async()` wrapping.

Reply:
I'm not sure aiofiles is actually what we really want, it just uses a threadpool to do the actual I/O:

> aiofiles helps with this by introducing asynchronous versions of files that support delegating operations to a separate thread pool.

(on the main description: https://pypi.org/project/aiofiles/)

This is basically what `sync_to_async` does so this doesn't seem like it cuts out the context switch :/

Reply:
It seems I keep confusing you, sorry for that. I don't want to anything more with the file-base engine. I think we should keep the current versions of async methods (that call sync methods) to avoid `sync_to_async()` wrapping that is in the base implementation.